src/Context/Course/Module/Course/Application/Create/CourseCreator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace CodelyTv\Context\Course\Module\Course\Application\Create;
+
+use CodelyTv\Context\Course\Module\Course\Domain\Course;
+use CodelyTv\Context\Course\Module\Course\Domain\CourseDescription;
+use CodelyTv\Context\Course\Module\Course\Domain\CourseRepository;
+use CodelyTv\Context\Course\Module\Course\Domain\CourseTitle;
+use CodelyTv\Shared\Domain\Bus\Event\DomainEventPublisher;
+use CodelyTv\Shared\Domain\CourseId;
+
+final class CourseCreator
+{
+    private $repository;
+    private $publisher;
+
+    public function __construct(CourseRepository $repository, DomainEventPublisher $publisher)
+    {
+        $this->repository = $repository;
+        $this->publisher = $publisher;
+    }
+
+    public function create(CourseId $id, CourseTitle $title, CourseDescription $description)
+    {
+        $course = new Course($id, $title, $description);
```

Comment:
Better to use a named constructor as in https://github.com/CodelyTV/cqrs-ddd-php-example/blob/d3da9074b765d0612e4c467e54d0cac94ba7d7c9/src/Context/Video/Module/Video/Application/Create/VideoCreator.php#L26

Why: https://codely.tv/screencasts/constructores-semanticos/ + record events inside the named constructor leaving the default constructor for reconstructing `Video` instances in a certain state without recording these events.